components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java

```diff
@@ -1437,6 +1438,25 @@ EnvironmentPropertiesDTO getEnvironmentSpecificAPIProperties(String apiUuid, Str
      */
     Environment getEnvironment(String organization, String uuid) throws APIManagementException;
 
+    /**
+     * Get Endpoint details by providing API UUID.
+     *
+     * @param uuid      Unique Identifier of API
+     * @return List<APIEndpointInfo> Object
+     * @throws APIManagementException if an error occurs while retrieving revision details
+     */
+    List<APIEndpointInfo> getAllAPIEndpointsByUUID(String uuid) throws APIManagementException;
+
+    /**
+     * Get Endpoint details by providing API UUID.
+     *
+     * @param apiUUID      Unique Identifier of API
+     * @param endpointUUID      Unique Identifier of Endpoint
+     * @return APIEndpointInfo Object
+     * @throws APIManagementException if an error occurs while retrieving revision details
+     */
+    APIEndpointInfo getAPIEndpointByUUID(String apiUUID, String endpointUUID) throws APIManagementException;
```

Comment:
Is it necessary to use apiUUID when retrieving an endpoint. I think endpointUUID might be enough?

Reply:
we need the revisionId actually. endpointUUID is not changing for each revision i think.